osquery/utils/darwin/plist.mm

```diff
@@ -175,27 +175,30 @@ Status parsePlistContent(const std::string& content, pt::ptree& tree) {
 Status parsePlist(const fs::path& path, pt::ptree& tree) {
   tree.clear();
 
-  auto status = Status();
+  std::string plist_content;
+  auto status = readFile(path.string(), plist_content);
 
-  @autoreleasepool {
-    id ns_path = [NSString stringWithUTF8String:path.string().c_str()];
-    id stream = [NSInputStream inputStreamWithFileAtPath:ns_path];
-    if (stream == nil) {
-      return Status(1, "Unable to read plist: " + path.string());
-    }
+  if (!status.ok()) {
+    return Status::failure("Unable to read plist: " + path.string());
+  }
+
+  NSData* plist_content_view =
+      [NSData dataWithBytesNoCopy:plist_content.data()
+                           length:plist_content.length()
+                     freeWhenDone:false];
 
+  @autoreleasepool {
     // Read file content into a data object, containing potential plist data.
     NSError* error = nil;
-    [stream open];
-    id plist_data = [NSPropertyListSerialization propertyListWithStream:stream
-                                                                options:0
-                                                                 format:NULL
-                                                                  error:&error];
+    id plist_data =
+        [NSPropertyListSerialization propertyListWithData:plist_content_view
+                                                  options:0
```

Comment:
very minor nit, feel free to ignore -- for `options`, prefer explicit value of `NSPropertyListImmutable` instead of `0` (it's the same as zero, but imo makes it a bit easier to read)

Reply:
fixed, thanks! This slipped through the cracks.